### PR TITLE
fix(angular1_router): generate hash-bang links correctly

### PR DESCRIPTION
**Pull request by petebacondarwin**
See #7423

### CODE REVIEW

File: modules/angular1_router/lib/facades.es5 [hunk 1]
```diff
@@ -316,3 +316,13 @@ Location.prototype.path = function () {
 Location.prototype.go = function (path, query) {
   return $location.url(path + query);
 };
+Location.prototype.prepareExternalUrl = function(url) {
+  if (url.length > 0 && !url.startsWith('/')) {
+    url = '/' + url;
+  }
+  if(!$location.$$html5) {
+    return '#' + url;
```
Comment (gkalpak) [inline modules/angular1_router/lib/facades.es5]: What about `hashPrefix` ?

Comment (petebacondarwin) [inline modules/angular1_router/lib/facades.es5]: Good spot

**Comment by petebacondarwin** [comment]
@davidreher & @mgol I have made some updates to the code and tests for this PR. Can you take look?

**Comment by davidreher** [comment]
LGTM

**Comment by btford** [comment]
LGTM!

**Pull request closed without merging**